### ISSUE TITLE
can nfd-worker runs on specific nodes?

### ISSUE THREAD

**Issue by wlane** [kubernetes-sigs/node-feature-discovery-operator#250]
Currently, the nfd-worker runs by default on all available worker nodes. Are there any user-defined parameters to specify which nodes it should run on? I only need it to run on certain nodes with a specific label. Thanks.

**Comment by marquiz** [comment]
@yevgeny-shnaidman does the operator have possibility to set nodeSelector for the worker?

/remove-lifecycle stale

**Comment by yevgeny-shnaidman** [comment]
no, we don''t have any such field in the CRD. We can add it , if needed, but on the other hand, does it really matter if they run on all the nodes? It does not interfere with the other functionalities, and the footprint is not not big

**Comment by yevgeny-shnaidman** [comment]
@if this is the only issue, then i think the better solution it to make operator add the node selector labels to the worker DS based on arch type label without any interference from the CRD

**Comment by bernhardloos** [comment]
That would be enough from my use-case and a lot simpler.

**Comment by marquiz** [comment]
@yevgeny-shnaidman I think this is still an open issue(?)

/remove-lifecycle rotten

**Comment by marquiz** [comment]
@yevgeny-shnaidman oh yes #261, I had already forgotten that. Let's close this now as fixed/implemented
(cc @bernhardloos)

/close
